File: swarmmanagercontroller/src/main/java/co/uk/swarmbit/docker/api/configs/ConfigsApi.java
package co.uk.swarmbit.docker.api.configs;

import co.uk.swarmbit.docker.api.configs.parameters.ConfigsListParameters;
import co.uk.swarmbit.docker.api.common.json.ConfigCreateResponseJson;
import co.uk.swarmbit.docker.api.common.json.ConfigJson;
import co.uk.swarmbit.docker.api.configs.parameters.ConfigsCreateParameters;
import co.uk.swarmbit.docker.api.configs.parameters.ConfigsUpdateParameters;

import java.util.List;

public interface ConfigsApi {

    List<ConfigJson> listConfigs(String swarmId, ConfigsListParameters parameters);

    ConfigJson inspectConfig(String swarmId, String id);

    ConfigCreateResponseJson createConfig(String swarmId, ConfigsCreateParameters parameters);

    void deleteConfig(String swarmId, String id);

    void updateConfig(String swarmId, String id, ConfigsUpdateParameters parameters);
}

File: swarmmanagercontroller-new/src/main/kotlin/com/swarmbit/docker/api/configs/ConfigsApi.kt
package com.swarmbit.docker.api.configs

import com.swarmbit.docker.api.configs.model.Config
import com.swarmbit.docker.api.configs.model.ConfigCreateResponse
import com.swarmbit.docker.api.configs.parameters.ConfigsCreateParameters
import com.swarmbit.docker.api.configs.parameters.ConfigsListParameters
import com.swarmbit.docker.api.configs.parameters.ConfigsUpdateParameters

interface ConfigsApi {
    fun listConfigs(swarmId: String, parameters: ConfigsListParameters = ConfigsListParameters()): List<Config>
    fun inspectConfig(swarmId: String, id: String): Config
    fun createConfig(swarmId: String, parameters: ConfigsCreateParameters): ConfigCreateResponse
    fun deleteConfig(swarmId: String, id: String)
    fun updateConfig(swarmId: String, id: String, parameters: ConfigsUpdateParameters)
}
